src/components/common/GProfilerModal.tsx

```diff
@@ -0,0 +1,335 @@
+                                                );
+                                            }
+                                        )}
+                                    </Table.Body>
+                                </Table>
+                                <Form.Group widths='equal'>
+                                    <Form.Field
+                                        control={Select}
+                                        label='Organism'
+                                        options={organisms}
+                                        placeholder='Choose an organism'
+                                        onChange={handleSelectOrganism}
+                                        disabled={
+                                            gProfilerToken !== null &&
+                                            gProfilerToken !== ''
+                                                ? true
+                                                : false
+                                        }
+                                        value={selectedOrganism}
+                                    />
+                                    <Form.Field
+                                        control={Input}
+                                        label='g:Profiler Token (Optional)'
+                                        placeholder='Token'
+                                        value={this.state.gProfilerToken}
+                                        onChange={handleChangeToken}
+                                    />
+                                </Form.Group>
+                                {this.state.error !== null && (
+                                    <Form.Group>
+                                        <Label basic color='red'>
+                                            {this.state.error}
+                                        </Label>
+                                    </Form.Group>
+                                )}
+                            </Form>
+                        </Modal.Description>
+                    </Modal.Content>
+                    <Modal.Actions>
+                        {
+                            <Button
+                                type='button'
+                                value='create-gprofiler-link'
+                                onClick={handleClickCreateGProfilerLink}
+                                secondary>
+                                {'Create Link'}
+                            </Button>
+                        }
+                        {this.state.gProfilerURL !== null && (
+                            <Button
+                                type='button'
+                                value='goto-gprofiler'
+                                onClick={handleClickGotoGProfilerURL}
+                                primary>
+                                {'Go to g:Profiler'}
+                            </Button>
+                        )}
```

Comment:
This might be nicer if the button changes to the the one for the link, rather than a second button appearing, a little more intuitive maybe.

Reply:
Not bad idea, I'll change it to your suggestion